photon-api/src/main/scala/com/linkedin/photon/ml/evaluation/MultiEvaluator.scala

```diff
@@ -16,19 +16,21 @@ package com.linkedin.photon.ml.evaluation
 
 import org.apache.spark.rdd.RDD
 
+import com.linkedin.photon.ml.Types.UniqueSampleId
+
 /**
- * Evaluator sharded with the specified ids.
+ * Evaluator applied to a collection of samples grouped by some ID.
```

Comment:
Add a big comment, explaining that tagId is e.g. "songId", and that a MultiEval will take that tagId, find out that "song123", "song456" have that tagId, then find all the training samples that contain "song123", "song456" and compute <something> with that then do some averaging ... or something similarly appropriate, with an example.